openwisp_network_topology/visualizer/views.py

```diff
@@ -26,6 +34,17 @@ class TopologyDetailView(View, GraphVisualizerUrls):
 
     def get(self, request, pk):
         topology = get_object_or_404(self.topology_model, pk)
+        user = request.user
+        if TOPOLOGY_API_AUTH_REQUIRED and not user.is_superuser:
+            auth_perm = check_auth_perm(request)
+            if auth_perm is not None:
+                return auth_perm
```

Comment:
I opened this view as an unauthenticated user and I was able to access it anyway

Reply:
@nemesisdesign check well. this works well on my machine and on `travis CI` since the test cases are passing for the different django and python versions. either you are not using this branch or you have not turned on `TOPOLOGY_API_AUTH_REQUIRED`.